frontend/src/metabase/admin/settings/components/widgets/EmbeddingOption/SwitchWithSetByEnvVar/SwitchWithSetByEnvVar.unit.spec.tsx

```diff
@@ -0,0 +1,156 @@
+import userEvent from "@testing-library/user-event";
+import fetchMock from "fetch-mock";
+
+import {
+  setupPropertiesEndpoints,
+  setupSettingsEndpoints,
+} from "__support__/server-mocks";
+import { mockSettings } from "__support__/settings";
+import { renderWithProviders, screen } from "__support__/ui";
+import type { SettingDefinition } from "metabase-types/api";
+import {
+  createMockSettingDefinition,
+  createMockSettings,
+} from "metabase-types/api/mocks";
+import {
+  createMockAdminState,
+  createMockState,
+} from "metabase-types/store/mocks";
+
+import {
+  SwitchWithSetByEnvVar,
+  type SwitchWithSetByEnvVarProps,
+} from "./SwitchWithSetByEnvVar";
+
+type SetupProps = Partial<
+  Pick<SwitchWithSetByEnvVarProps, "settingKey" | "label" | "disabled">
+> &
+  Omit<SettingDefinition<SwitchWithSetByEnvVarProps["settingKey"]>, "key">;
+
+const setup = ({
+  settingKey = "enable-embedding-static",
```

Comment:
I'd remove any props that you don't pass in the test and hardcode them, so it's easier to understand what we are passing.

Reply:
removed those props